File: CaesarCipher/CaesarCipher.scala
def ReturnEncKey(num: Int, key: Int) : Int = {
	return (num+key)%26;
}

def ReadPlainText(key : Int) : Unit = {
	val stringBuilder = StringBuilder()
	val lower_chars = ('a' to 'z').zipWithIndex
	val upper_chars = ('A' to 'Z').zipWithIndex
	print("Enter the plaintext: ");
	var word = scala.io.StdIn.readLine();
	for (char <- word.toList) {
		if (char.isUpper) {
			for ((alph, index) <- upper_chars) {
				if (alph==char) {
					var enc_key = ReturnEncKey(index, key)
					var enc_char = upper_chars(enc_key)._1
					stringBuilder.append(enc_char)
				}
			}
		}
		else {
			for((alph, index) <- lower_chars) {
				if (alph==char) {
					var enc_key = ReturnEncKey(index, key)
					var enc_char = lower_chars(enc_key)._1
					stringBuilder.append(enc_char)
				}
			}
		}
	}
	val resultString: String = stringBuilder.toString
	print(s"Encrypted Text: $resultString")
}

@main def main(): Unit =
	print("Enter Key: ")
	var key = scala.io.StdIn.readInt();
	//print(key);
	if (1<=key && key<=25) {
		ReadPlainText(key)		
	}
	else {
		print("Invalid Key")
	}

File: CaesarCipher/CaesarCipher.js
function return_enc_key(key, index) {
  const num = index + (key % 26);
  return num;
}

function read_plaintext(key) {
  var result = "";
  const alphabets = [];
  for (let i = 65; i <= 90; i++) {
    alphabets.push(String.fromCharCode(i));
  }
  var plain_text = prompt("Enter a plain text: ");
  var plain_text_upper = plain_text.toUpperCase();
  const plain_array = plain_text_upper.split("");
  for (let i = 0; i < plain_array.length; i++) {
    for (let j = 0; j < alphabets.length; j++) {
      if (plain_array[i] === alphabets[j]) {
        const plains = plain_array[i];
        var enc_key = return_enc_key(key, alphabets.indexOf(plains));
        result += alphabets[enc_key];
      }
    }
  }
  console.log("Encrypted Text: " + result);
}

function read_input() {
  var key = prompt("Enter a Key: ");
  if (1 <= key && key <= 25) {
    read_plaintext(key);
  } else {
    console.log("Enter a Valid Key");
  }
}

read_input();
